app/components/onboarding_channels_checkboxes/onboarding_channels_checkboxes.html.erb

```diff
@@ -7,10 +7,25 @@
   </p>
   <ul class="OnboardingChannelsCheckboxes-list">
     <% Setting.channels.each do |key, value| %>
-      <li class="OnboardingChannelsCheckboxes-listItem">
-        <%= c 'checkbox', id: "setting[channels][#{key}]", checked: value %>
-        <label for=<%= "setting[channels][#{key}]" %>><%= key.to_s.camelize %>
-      </li>
+      <input
+        id='<%= "setting[channels][#{key}][configured]" %>'
+        name='<%= "setting[channels][#{key}][configured]" %>'
+        value="<%= value[:configured] %>"
+        type="hidden"
+      >
+      <% if value[:configured] %>
+        <li class="OnboardingChannelsCheckboxes-listItem">
+          <%= c 'checkbox', id: "setting[channels][#{key}][allow_onboarding]", checked: value[:allow_onboarding] %>
+          <label for=<%= "setting[channels][#{key}][allow_onboarding]" %>><%= key.to_s.camelize %>
+        </li>
+      <% else %>
+        <input
+          id='<%= "setting[channels][#{key}][allow_onboarding]" %>'
+          name='<%= "setting[channels][#{key}][allow_onboarding]" %>'
+          value="<%= value[:allow_onboarding] %>"
+          type="hidden"
+        >
```

Comment:
I don't know if this is the best approach because hidden inputs are still available in the dom tree and could be modified by a malicious user. In this case, this is not a public website and this section is only visible to admin, so I think it's ok.

The reason this conditional exists is because I want to have the entire hash saved in Setting.channels sent from the form to the controller because when the update occurs it simply replaces the entire has with a potentially modified one.